numcodecs/zarr3.py

```diff
@@ -0,0 +1,275 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, replace
+from functools import cached_property
+import math
+from typing_extensions import Self
+from warnings import warn
+
+import numpy as np
+import numcodecs
+
```

Comment:
Might be nice to have a try: import zarr; assert zarr.version=3; except: raise ImportError("zarr is required to use numcodecs zarr integration block (sorry for my awful pseudocode), so there's a nice error if a user tries to import but doesn't have zarr>=3 installed.